frontend/appflowy_tauri/src/appflowy_app/components/_shared/database-hooks/useCell.ts

```diff
@@ -0,0 +1,40 @@
+import { CellIdentifier } from '../../../stores/effects/database/cell/cell_bd_svc';
+import { CellCache } from '../../../stores/effects/database/cell/cell_cache';
+import { FieldController } from '../../../stores/effects/database/field/field_controller';
+import { CellControllerBuilder } from '../../../stores/effects/database/cell/controller_builder';
+import { DateCellDataPB, FieldType, SelectOptionCellDataPB } from '../../../../services/backend';
+import { useState } from 'react';
+
+export const useCell = (cellIdentifier: CellIdentifier, cellCache: CellCache, fieldController: FieldController) => {
+  const [data, setData] = useState<string[]>([]);
+
+  const loadCell = async () => {
+    const builder = new CellControllerBuilder(cellIdentifier, cellCache, fieldController);
+    const cellController = builder.build();
+    cellController.subscribeChanged({
```

Comment:
I think this cell controller should be used in the corresponding field-type cell components. Each component uses useEffect to subscribe to the change and returns an unsubscribe action to the effect when the current component gets unmounted.

Reply:
Agreed. I've created component for each type and moved useCell inside of them